utilities/transactions/pessimistic_transaction_db.cc

```diff
@@ -742,6 +742,26 @@ Status PessimisticTransactionDB::GetTimestampedSnapshots(
   return db_impl_->GetTimestampedSnapshots(ts_lb, ts_ub, timestamped_snapshots);
 }
 
+Status PessimisticTransactionDB::CollapseKey(const Slice& key,
+                                             const WriteOptions& writeOptions,
+                                             const ReadOptions& readOptions) {
+  auto tx = std::unique_ptr<Transaction>(
+      BeginTransaction(writeOptions, TransactionOptions(), nullptr));
```

Comment:
For now it might be easier to make this a `Transaction` function. That way you don't need to take `WriteOptions` or `TransactionOptions`. Plus it'll allow the user to combine this work with other things in the same transaction